.github/workflows/test-docker-image-build.yml

```diff
@@ -0,0 +1,23 @@
+# https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions
+
+name: Test Docker image build
```

Comment:
Is the idea here to just check to make sure the docker image can be built? This just seems to build the image and then forget about it.

Reply:
I think this is the intent. Does it block the PR or merging if it fails?

Reply:
Confirmed. The idea here is to make sure the docker image can be built, and _**if not then to block the PR**_.

I didn't actually check if the failure really blocks the PR, I will do it in my [side draft PR](https://github.com/microsoft/Quantum/pull/789). Good point.

Reply:
Confirmed, the [failure does block](https://github.com/microsoft/Quantum/pull/789) the PR.